Mage/Command/Factory.php

```diff
@@ -43,14 +43,17 @@ public static function get($commandName, Config $config)
             // try a custom command
             $className = 'Command\\' . $commandName;
 
+            // TODO use a custom exception
             if (!class_exists($className)) {
                 throw new Exception('Command "' . $commandName . '" not found.');
             }
         }
 
         /** @var AbstractCommand $instance */
+        // TODO dependencies like $config should be injected into constructor
```

Comment:
No, I meant some DI mechanism, like Sf2 Dependency Injection, or any other, loaded by composer dependency that we could use. We can, of course, write our own DI.
Issue I mentioned about is: https://github.com/andres-montanez/Magallanes/issues/88
It's only short sentence but, I don't know why, I keep that in mind.